toddredmond/homework.md

```diff
@@ -0,0 +1,161 @@
+circumference<=area or (diameter*π)<=((l+r)*(t+b))
+
+**d)**
+Write an expression for the area of the smallest circle which completely encloses (i.e. circumscribes) the rectangle.
+((sqrt(a2+b2))*π)==((l+r)*(t+b))
+
+**e)**
+Imagine subdividing your rectangle into 3 equal rows and 3 equal columns, which would create 9 smaller rectangles, identical in shape but varying by position.
+Define four new variables describing the centermost small rectangle.
+(_Hint: one of the many solutions is very similar to the solution of **2b** above._)
+var row, column;
+row = 2;
+column = 2;
+center = row && column;
+
+
+**4)** (_Difficulty: easyish_)
+
+Imagine that the squares of an ordinary checkerboard are numbered in two different ways:
+
+* Each square has integer coordinates _(R,C)_ describing its row and column.  Both values should be in the range 0..7, so that the upper-left square is at (0,0) and the bottom-right is at (7,7).
+
+* Each square has a unique integer number N from 0 to 63.  These numbers run sequentially left-to-right one row at a time, top to bottom.  Therefore the upper-left square has N==0 and the bottom-right has N==63.
+
+**a)**  Given a particular R and C, find the corresponding N.  That is, write an expression for variable N containing variables R and C. N = (R*8)+C
+
+**b)**  Given N, find R.  Write an expression for R which contains N. R = (N-R)/8
+
+**c)**  Given N, find C.  Write an expression for C which contains N. C = N-(R*8)
+
+**d)**  Assume the squares are colored black and white, with the upper-left square black.
+Write an expression to set a variable _color_ to either 'black' or 'white', describing the square identified by variables R,C, and N.
+(_Hint: If you prefer, you may use a pair of conditional statements instead of a single expression._)
+var counter = 0, var color;
+while (counter <= 63) {
+  if (counter%2===0) {
+    color = Black;
+  }
+  else {
+    color = White;
+  };
+  counter = counter + 1;
+}
+---
+
+**5)** (_Difficulty: moderate_)
+
+Suppose you represent a fraction (_n/d_) with 2 integer variables: _n_ for the numerator and _d_ for the denominator.
+If _n_ is greater than _d_, the fraction is "improper", but it can be rewritten as a proper fraction.  For example, "7/4" is improper, but it can be rewritten as "1 3/4", which is proper.
+
+Write an expression for a string expressing the proper form of an improper fraction _n/d_.  For example, when _n==7_ and _d==4_, your resulting string should be "1 3/4".  You may assume both _n_ and _d_ are positive integers and _n_ > _d_, but otherwise you should be able to handle any values of _n_ and _d_.
+(_Hint: you'll need the modulo operator _%_, and you'll probably want to create a couple of extra variables._) 
+var n=7, d=4, result;
+result = n/d;
+result = result%1;
+result = result*100;
+result = Math.round(result);
+common = 100/d;
+result = Math.round(result/common);
+dem = 100/common;
+fraction = result + "/" + dem;
+console.log("1 "+ fraction);
```

Comment:
This one gets trickier. Style-wise, rather than a bunch of redefinitions of the same variable, it might be more useful to you (and also easier to see what's going on) if you name each step. For example, you could say var remainder = n%d; Naming things is a huge part of coding--code with intuitive names is easy to read through and barely needs comments, while code with more generic names (like "result", I'm afraid to say) can be more difficult.

You'll also want to reevaluate this one and check to make sure that it works in a number of different cases--while the fraction was working out for me, the whole number seems to always come up as 1.